### ISSUE TITLE
java.lang.NullPointerException during mvn org.owasp:dependency-check-maven:aggregate

### ISSUE THREAD

**Issue by somera** [dependency-check/DependencyCheck#6913]
**Describe the bug**
Today I updated spring-boot dependency
```
    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>3.2.9</version>
        <relativePath />
    </parent>
```
in my project and after I run `mvn org.owasp:dependency-check-maven:aggregate` I saw this:
```
[INFO] --- dependency-check:10.0.3:aggregate (default-cli) @ xxxxxxx ---
[WARNING] The artifact com.github.wnameless:json-flattener:jar:0.7.1 has been relocated to com.github.wnameless.json:json-flattener:jar:0.7.1
[INFO] Checking for updates
[INFO] NVD API has 233 records in this update
[INFO] Downloaded 233/233 (100%)
[ERROR] Failed to process CVE-2024-8033
java.lang.NullPointerException
    at java.util.stream.ReferencePipeline$7$1.accept (ReferencePipeline.java:273)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
    at java.util.ArrayList$ArrayListSpliterator.tryAdvance (ArrayList.java:1602)
    at java.util.stream.ReferencePipeline$7$1.accept (ReferencePipeline.java:280)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
    at java.util.ArrayList$ArrayListSpliterator.tryAdvance (ArrayList.java:1602)
    at java.util.stream.ReferencePipeline.forEachWithCancel (ReferencePipeline.java:129)
    at java.util.stream.AbstractPipeline.copyIntoWithCancel (AbstractPipeline.java:527)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:513)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:499)
    at java.util.stream.MatchOps$MatchOp.evaluateSequential (MatchOps.java:230)
    at java.util.stream.MatchOps$MatchOp.evaluateSequential (MatchOps.java:196)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.anyMatch (ReferencePipeline.java:632)
    at org.owasp.dependencycheck.data.nvdcve.CveItemOperator.testCveCpeStartWithFilter (CveItemOperator.java:228)
    at org.owasp.dependencycheck.data.nvdcve.CveDB.updateVulnerability (CveDB.java:1098)
    at org.owasp.dependencycheck.data.update.nvd.api.NvdApiProcessor.updateCveDb (NvdApiProcessor.java:119)
    at org.owasp.dependencycheck.data.update.nvd.api.NvdApiProcessor.call (NvdApiProcessor.java:96)
    at org.owasp.dependencycheck.data.update.nvd.api.NvdApiProcessor.call (NvdApiProcessor.java:40)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:840)
[INFO] Completed processing batch 1/1 (100%) in 482ms
[INFO] Updating CISA Known Exploited Vulnerability list: https://www.cisa.gov/sites/default/files/feeds/known_exploited_vulnerabilities.json
[INFO] Begin database defrag
[INFO] End database defrag (6187 ms)
[INFO] Check for updates complete (10451 ms)
[INFO] 
```
No NPE was seen in the second run.

**Version of dependency-check used**
mvn org.owasp:dependency-check-maven:aggregate
```
                    <groupId>org.owasp</groupId>
                    <artifactId>dependency-check-maven</artifactId>
                    <version>10.0.3</version>
```

**To Reproduce**
I can't reproduce it.

**Expected behavior**
No NPE.

**Comment by somera** [comment]
Same problem on other computer
```
20:40:35  [INFO] Checking for updates
20:40:40  [INFO] NVD API has 697 records in this update
20:40:41  [INFO] Downloaded 697/697 (100%)
20:40:45  [ERROR] Failed to process CVE-2024-8033
20:40:45  java.lang.NullPointerException
20:40:45      at java.util.stream.ReferencePipeline$7$1.accept (ReferencePipeline.java:273)
20:40:45      at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
20:40:45      at java.util.ArrayList$ArrayListSpliterator.tryAdvance (ArrayList.java:1602)
20:40:45      at java.util.stream.ReferencePipeline$7$1.accept (ReferencePipeline.java:280)
20:40:45      at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
20:40:45      at java.util.ArrayList$ArrayListSpliterator.tryAdvance (ArrayList.java:1602)
20:40:45      at java.util.stream.ReferencePipeline.forEachWithCancel (ReferencePipeline.java:129)
20:40:45      at java.util.stream.AbstractPipeline.copyIntoWithCancel (AbstractPipeline.java:527)
20:40:45      at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:513)
20:40:45      at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:499)
20:40:45      at java.util.stream.MatchOps$MatchOp.evaluateSequential (MatchOps.java:230)
20:40:45      at java.util.stream.MatchOps$MatchOp.evaluateSequential (MatchOps.java:196)
20:40:45      at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
20:40:45      at java.util.stream.ReferencePipeline.anyMatch (ReferencePipeline.java:632)
20:40:45      at org.owasp.dependencycheck.data.nvdcve.CveItemOperator.testCveCpeStartWithFilter (CveItemOperator.java:228)
20:40:45      at org.owasp.dependencycheck.data.nvdcve.CveDB.updateVulnerability (CveDB.java:1098)
20:40:45      at org.owasp.dependencycheck.data.update.nvd.api.NvdApiProcessor.updateCveDb (NvdApiProcessor.java:119)
20:40:45      at org.owasp.dependencycheck.data.update.nvd.api.NvdApiProcessor.call (NvdApiProcessor.java:96)
20:40:45      at org.owasp.dependencycheck.data.update.nvd.api.NvdApiProcessor.call (NvdApiProcessor.java:40)
20:40:45      at java.util.concurrent.FutureTask.run (FutureTask.java:264)
20:40:45      at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
20:40:45      at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
20:40:45      at java.lang.Thread.run (Thread.java:840)
20:40:45  [INFO] Completed processing batch 1/1 (100%) in 4.276ms
```

**Comment by TIBCOrkrajews** [comment]
Definitely seeing this.

**Comment by aikebah** [comment]
Initial assessment looking at 
https://nvd.nist.gov/vuln/detail/CVE-2024-8033
there is at the moment some unanticipated data error in the NVD database

CPE Configuration 		
```
AND
     OR
     OR
          cpe:2.3:o:microsoft:windows:-:*:*:*:*:*:*:*
```

leading up to a null-valued CPEMatch node due to the NVD API not abiding by its [defined schema](https://csrc.nist.gov/schema/nvd/api/2.0/cve_api_json_2.0.schema) that defines cpeMatch as a required member for each node in the configurations object (which the first node violates as it only has the operator and negate properties)

```json
        "configurations": [
          {
            "operator": "AND",
            "nodes": [
              {
                "operator": "OR",
                "negate": false
              },
              {
                "operator": "OR",
                "negate": false,
                "cpeMatch": [
                  {
                    "vulnerable": false,
                    "criteria": "cpe:2.3:o:microsoft:windows:-:*:*:*:*:*:*:*",
                    "matchCriteriaId": "A2572D17-1DE6-457B-99CC-64AFD54487EA"
                  }
                ]
              }
            ]
          }
        ],
```

**Comment by aikebah** [comment]
@jeremylong WDYT... should we fix this here as a consumer of the open vulnerabiltiy client, or is it better to prevent creation of `null`-valued cpeMatches in the Open-Vulnerability-Project?

**Comment by aikebah** [comment]
I've contacted NVD by mail to try and get this faulty data corrected on their side

**Comment by somera** [comment]
From the deelopment view a "corrupt" data should not trigger an NPE. But if the data will be fixed it's ok too.

And the question is what happens in the code? Was the NPE only for the one CVE? Were all other 696 CVEs processed?

**Comment by aikebah** [comment]
Fully agree that the code should be fixed to deal with it more gracefully than NPE

**Comment by aikebah** [comment]
> And the question is what happens in the code? Was the NPE only for the one CVE? Were all other 696 CVEs processed?

Was still an open item on my list to look into. The error is dealt with at the level of the single CVE record, so the effect of the error is that only that CVE gets not added/updated in the database. The other CVEs are successfully added/updated in the database.
The faulty CVE will not be processed again until NVD publishes new information in it (which makes it re-appear in the API response for 'CVEs added/updated after the last import of CVE data into the databse') or you purge your Owasp DC data (which would be also how you would be able to reproduce the current failure (at the cost of processing 260k+ CVE entries))

**Comment by jeremylong** [comment]
The fix should probably be in the open-vulnerability-client.  I'll put a PR together.

**Comment by jeremylong** [comment]
after looking at the OVC code - the fix is much easier to do in ODC; see #6915.
